src/main/java/org/broadinstitute/hellbender/tools/variantdb/ArrayExtractCohortEngine.java

```diff
@@ -0,0 +1,363 @@
+            } else {
+                probeId = (Long) sortedRow.get("probe_id");
+            }
+
+            if ( probeId != currentProbeId && currentProbeId != -1 ) {
+                ++totalNumberOfSites;
+                processSampleRecordsForLocation(currentProbeId, currentPositionRecords, columnNames);
+                currentPositionRecords.clear();
+            }
+
+            currentPositionRecords.add(sortedRow);
+            currentProbeId = probeId;
+        }
+
+        if ( ! currentPositionRecords.isEmpty() ) {
+            ++totalNumberOfSites;
+            processSampleRecordsForLocation(currentProbeId, currentPositionRecords, columnNames);
+        }
+    }
+
+    private void processSampleRecordsForLocation(final long probeId, final Iterable<GenericRecord> sampleRecordsAtPosition, final Set<String> columnNames) {
+        final List<VariantContext> unmergedCalls = new ArrayList<>();
+        final Set<String> currentPositionSamplesSeen = new HashSet<>();
+        boolean currentPositionHasVariant = false;
+
+        final ProbeInfo probeInfo = probeIdMap.get(probeId);
+        if (probeInfo == null) {
+            throw new RuntimeException("Unable to find probeInfo for " + probeId);
+        }
+
+        final String contig = probeInfo.contig;
+        final long position = probeInfo.position;
+        final Allele refAllele = Allele.create(refSource.queryAndPrefetch(contig, position, position).getBaseString(), true);
+
+        int numRecordsAtPosition = 0;
+
+        for ( final GenericRecord sampleRecord : sampleRecordsAtPosition ) {
+            final long sampleId = (Long) sampleRecord.get(SchemaUtils.SAMPLE_ID_FIELD_NAME);
+
+            // TODO: handle missing values
+            String sampleName = sampleIdMap.get((int) sampleId);            
+            currentPositionSamplesSeen.add(sampleName);
+
+            ++numRecordsAtPosition;
+
+            if ( printDebugInformation ) {
+                logger.info("\t" + contig + ":" + position + ": found record for sample " + sampleName + ": " + sampleRecord);
+            }
+
+            ++totalNumberOfVariants;
+            unmergedCalls.add(createVariantContextFromSampleRecord(probeInfo, sampleRecord, columnNames, contig, position, sampleName));
+
+        }
+
+        if ( printDebugInformation ) {
+            logger.info(contig + ":" + position + ": processed " + numRecordsAtPosition + " total sample records");
+        }
+
+        finalizeCurrentVariant(unmergedCalls, currentPositionSamplesSeen, contig, position, refAllele);
+    }
+
+    private void finalizeCurrentVariant(final List<VariantContext> unmergedCalls, final Set<String> currentVariantSamplesSeen, final String contig, final long start, final Allele refAllele) {
+
+        // TODO: this is where we infer missing data points... once we know what we want to drop
+        // final Set<String> samplesNotEncountered = Sets.difference(sampleNames, currentVariantSamplesSeen);
```

Comment:
do you think we are going to be dropping something we aren't already?

Reply:
maybe... we need to dig into more data, and when we get to imputed data I think we will